mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -201,6 +210,102 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
       }
     }
 
+    int64_t gemmM_size, gemmN_size, gemmK_size;
+    int64_t gemmM_extra, gemmN_extra, gemmK_extra;
+    gemmM_extra = gemmN_extra = gemmK_extra = 0;
+    // compute we should use extra padding kernel or not
+    // c,k already / g ,so we can skip / g here
+    if (convOpType == miopen::ConvOpType::Conv2DOpType) {
+      gemmM_size = k;
+      gemmK_size = c * y * x;
+      gemmN_size = n * ho * wo;
+
+      llvm::errs() << n << " " << ho << " " << wo << "\n";
+    } else if (convOpType == miopen::ConvOpType::Conv2DBwdDataOpType) {
+      gemmM_size = c;
+      gemmK_size = k * y * x;
+      gemmN_size = n * ho * wo;
+    } else if (convOpType == miopen::ConvOpType::Conv2DBwdWeightOpType) {
+      gemmM_size = k;
+      gemmK_size = n * ho * wo;
+      gemmN_size = c * y * x;
+    }
+
+    bool needExtraPad = false;
+    int numOfFailedConfigs = 0;
+    bool isXdlops = false;
+    auto xdlopsV2Attr = op->template getAttrOfType<BoolAttr>("xdlopsV2");
+    if (xdlopsV2Attr && xdlopsV2Attr.getValue() == true)
+      isXdlops = true;
+
+    if (!isXdlops) {
```

Comment:
Logic here are almost the same between non-XDLOPS and XDLOPS case, probably using a lambda to factor out common codes